### ISSUE TITLE
Add option for smaller backups using openhab-cli

### ISSUE THREAD

**Issue by DanielMalmgren** [openhab/openhab-distro#755]
Somewhere between 2.2 and 2.3 the default behaviour of "openhab-cli backup" was changed as to include tmp and cache directories, which makes the backup file a multitude bigger (mine went from 300kB to 90MB). I suggest the behaviour is changed back to the original with an optional argument for those who also want the tmp and cache directories.

I have added this on my installation, attaching diff files for openhab-cli and backup. This adds a --full argument to the commands.

Also see discussion here: [https://community.openhab.org/t/openhab-cli-backup-from-700kb-to-85mb/48266/1](url)

[backup.diff.txt](https://github.com/openhab/openhab-distro/files/2332084/backup.diff.txt)
[openhab-cli.diff.txt](https://github.com/openhab/openhab-distro/files/2332085/openhab-cli.diff.txt)

**Comment by BClark09** [comment]
Just a note for this issue, the `openhab-cli` command is for linuxpkg installs only. But the issue exists in the `$OPENHAB_HOME/bin/backup` script, which will effect Mac and manual installs too.

**Comment by DanielMalmgren** [comment]
Ok, I didn't know that. The changes I've done to openhab-cli is mostly aesthetic.

I'd also like to note that my current implementation is far from optimal. It still copies the entire userdata folder to the temporary directory and then deletes tmp and cache from there which obviously is loads of megabytes unnecessarily copied. The best thing would be to not include them from the beginning, but cp doesn't have any way of excluding files. I guess that can be solved by replacing the cp (on line 118) with something else. I'll get back if I find anything good.

**Comment by Dan1001** [comment]
rsync has an exclude option, but would have the undesirable effect of adding another dependency.

**Comment by DanielMalmgren** [comment]
Yep, we like to stick with things that we know are always there. I made a version now that fixes this with help of find, sed and xargs, I guess those are standard commands? Attaching a new diff 😄 

[backup.diff.txt](https://github.com/openhab/openhab-distro/files/2335065/backup.diff.txt)

**Comment by DanielMalmgren** [comment]
...except that I forgot to clean up my test stuff. There obviously shouldn't be a -v on the cp command on line 119. I will keep that as an exercise for the reader to remove...

**Comment by BClark09** [comment]
When running the command I got the directories and files of `/userdata/config` inside `$TempDir/userdata`, but looking at the line, I'm not sure how this is possible?

**Comment by BClark09** [comment]
Yes, that's much better and working more efficiently too.

Would you care to submit a PR?

**Comment by DanielMalmgren** [comment]
I'd be happy to! Never done that before though, is there any simple guide you can point me to?

Also I started thinking about the removal of the backups folder. Now it copies over it and then removes it. Would it make sense removing it in the same fashion which I'm now doing with tmp and cache?

**Comment by DanielMalmgren** [comment]
Ok, I fiddled around for a bit and I think I've made a proper PR now, it really wasn't hard at all ;-)

This one is only for the changes in backup though, not in openhab-cli (since it's in a completely different repo). Would I have to make another PR for that?

Link to PR: [https://github.com/DanielMalmgren/openhab-distro/pull/1](url)

**Comment by DanielMalmgren** [comment]
Ok, obviously I still have something to learn about git :-)

I _think_ I've done it the right way now...

**Comment by DanielMalmgren** [comment]
I just installed M5 and realized that my PR just contained the changes to bin/backup and not openhab-cli. So now the backup is small but with no way of getting it "full" when using openhab-cli. Guess I'll have to create another PR?

**Comment by kaikreuzer** [comment]
Indeed, openhab-cli was not part of https://github.com/openhab/openhab-distro/pull/756/files. So yes, please create another PR for it!

**Comment by BClark09** [comment]
I only just noticed this, submitted a PR now in openhab-linuxpkg. Thanks for the work here @DanielMalmgren !

**Comment by BClark09** [comment]
I think this issue can be closed now.
